src/input/input.jsx

```diff
@@ -76,14 +76,12 @@ const Input = ({
     />
   )
 
-  const validate = []
-  if (required) validate.push(validateRequired(requiredMessage))
   const field = (
     <Field
       className={classnames({ [styles.error]: !!error, [styles.field]: type !== 'checkbox' })}
       component={getComponent(type)}
       {...commonProps}
-      validate={validate}
+      validate={validate || (required ? validateRequired : [])}
```

Comment:
You should make `[]` a constant so it doesn't change reference every rerender. 
This is an optimisation issue.

Reply:
Yes, you're right, forgot that ^^